### PR TITLE
Require a username when running a database-affecting Job from the CLI

### PR DESCRIPTION
**Pull request by glennmatthews**
### Fixes: #944 

When a Job is run with `--commit`, it automatically executes in a `change_logging()` context. This context hard-requires that an appropriate request context (with an associated User account) exists. The `nautobot-server runjob` command was not providing any such request or User, causing jobs to error out when run with --commit as soon as they attempted to make any database changes (triggering change-logging to run).

Solution:
- Add an additional parameter, `--username <username>` to the `nautobot-server runjob` command, as we don't have any way to map the calling (OS) user account automatically to a Django User.
- When `--commit` is included in a `runjob` command, make `--username <username>` a mandatory additional parameter. (This is not a backwards-incompatible change, as `runjob --commit <job>` would always error out previously due to #944; now it will fail with a CommandError and explanatory message instead of throwing an uncaught AttributeError)
- When `runjob --username <username> <job>` (with or without `--commit`) is called, construct the request context making use of this user and pass it into the `enqueue_job` function as appropriate.
- Update documentation
- Add unit testing.

```
# nautobot-server runjob --commit plugins/dummy_plugin.jobs/DummyJob
CommandError: --username is mandatory when running with --commit as True
```

```
# nautobot-server runjob --help
usage: nautobot-server runjob [-h] [--commit] [-u USERNAME] [--version] [-v {0,1,2,3}] [--settings SETTINGS]
                              [--pythonpath PYTHONPATH] [--traceback] [--no-color] [--force-color] [--skip-checks]
                              job

Run a job (script, report) to validate or update data in Nautobot

positional arguments:
  job                   Job to run

optional arguments:
  -h, --help            show this help message and exit
  --commit              Commit changes to DB (defaults to dry-run if unset). --username is mandatory if using this
                        argument
  -u USERNAME, --username USERNAME
                        User account to impersonate as the requester of this job
  --version             show program's version number and exit
  -v {0,1,2,3}, --verbosity {0,1,2,3}
                        Verbosity level; 0=minimal output, 1=normal output, 2=verbose output, 3=very verbose output
  --settings SETTINGS   The Python path to a settings module, e.g. "myproject.settings.main". If this isn't provided,
                        the DJANGO_SETTINGS_MODULE environment variable will be used.
  --pythonpath PYTHONPATH
                        A directory to add to the Python path, e.g. "/home/djangoprojects/myproject".
  --traceback           Raise on CommandError exceptions
  --no-color            Don't colorize the command output.
  --force-color         Force colorization of the command output.
  --skip-checks         Skip system checks.
```

### CODE REVIEW

File: nautobot/docs/additional-features/jobs.md [hunk 1]
```diff
@@ -390,7 +390,7 @@ When providing input data, it is possible to specify complex values contained in
 Jobs that do not require user input can be run from the CLI by invoking the management command:
 
 ```no-highlight
-nautobot-server runjob <grouping_name>/<module>/<JobName> [--commit]
+nautobot-server runjob [--username <username>] [--commit] <grouping_name>/<module>/<JobName>
```
Comment (hellerve) [inline nautobot/docs/additional-features/jobs.md:393]: Should we mention here as well that `--commit` requires `--username`?

File: nautobot/extras/management/commands/runjob.py [hunk 2]
```diff
@@ -27,6 +36,23 @@ def handle(self, *args, **options):
         if not job_class:
             raise CommandError('Job "%s" not found' % options["job"])
 
+        user = None
+        request = None
+        if options["commit"] and not options["username"]:
+            # Job execution with commit=True uses change_logging(), which requires a user as the author of any changes
+            raise CommandError("--username is mandatory when running with --commit as True")
```
Comment (hellerve) [inline nautobot/extras/management/commands/runjob.py:43]:
```suggestion
            raise CommandError("--username is mandatory when --commit is used")
```

File: nautobot/extras/management/commands/runjob.py [hunk 3]
```diff
@@ -27,6 +36,23 @@ def handle(self, *args, **options):
...
+
+        if options["username"]:
+            User = get_user_model()
+            try:
+                user = User.objects.get(username=options["username"])
+            except User.DoesNotExist as exc:
+                raise CommandError("No such username") from exc
```
Comment (hellerve) [inline nautobot/extras/management/commands/runjob.py:50]:
```suggestion
                raise CommandError("No such user") from exc
```

**Review by hellerve — APPROVED** [review]
Great work!

**Review by jathanism — APPROVED** [review]
This is a thing of beauty.

**Pull request merged**